csharp/src/ProtocolBuffers/Descriptors/FileDescriptor.cs

```diff
@@ -54,6 +54,22 @@ public sealed class FileDescriptor : IDescriptor<FileDescriptorProto>
         private readonly IList<FileDescriptor> publicDependencies;
         private readonly DescriptorPool pool;
 
+        public enum Syntax
+        {
+            UNKNOWN,
+            PROTO2,
+            PROTO3
+        }
+
+        public Syntax GetSyntax()

```

Comment:
This should be a property rather than a method:

public Syntax Syntax { get { return proto.Syntax == "proto3" ? Syntax.Proto3 : Syntax.Proto2; } }